google/downloader.sh

```diff
@@ -6,58 +6,100 @@
+    txt="$(dig TXT _netblocks${idx}.google.com +short @8.8.8.8 || true)"
+    ((idx++))
+  done
+}
 
-# get from netblocks
-txt="$(dig TXT _netblocks.google.com +short @8.8.8.8)"
-idx=2
-while [[ -n "${txt}" ]]; do
-  echo "${txt}" | tr '[:space:]+' "\n" | grep ':' | cut -d: -f2- >> /tmp/netblocks.txt
-  txt="$(dig TXT _netblocks${idx}.google.com +short @8.8.8.8)"
-  ((idx++))
-done
+fetch_netblocks
 
-# get from other netblocks
+# Function to resolve DNS SPF records recursively with validation
 get_dns_spf() {
    dig @8.8.8.8 +short txt "$1" |
    tr ' ' '\n' |
-   while read entry; do
+   while read -r entry; do
       case "$entry" in
-             ip4:*) echo "${entry#*:}" ;;
-             ip6:*) echo "${entry#*:}" ;;
+         ip4:*) echo "${entry#*:}" ;;
+         ip6:*) echo "${entry#*:}" ;;
          include:*) get_dns_spf "${entry#*:}" ;;
       esac
-   done
+   done || {
+     echo "Error: Failed to fetch DNS SPF records for $1"
+     exit 1
+   }
 }
 
-get_dns_spf "_cloud-netblocks.googleusercontent.com" >> /tmp/netblocks.txt
-get_dns_spf "_spf.google.com" >> /tmp/netblocks.txt
+# Fetch additional SPF-based netblocks with error handling
+get_dns_spf "_cloud-netblocks.googleusercontent.com" >> "$temp_dir/netblocks.txt"
+get_dns_spf "_spf.google.com" >> "$temp_dir/netblocks.txt"
 
+# Separate IPv4 and IPv6 ranges
+grep -v ':' "$temp_dir/goog.txt" > "$temp_dir/google-ipv4.txt"
+jq -r '.prefixes[] | select(.ipv4Prefix != null) | .ipv4Prefix' "$temp_dir/cloud.json" >> "$temp_dir/google-ipv4.txt"
+jq -r '.prefixes[] | select(.ipv4Prefix != null) | .ipv4Prefix' "$temp_dir/googlebot.json" >> "$temp_dir/google-ipv4.txt"
+grep -v ':' "$temp_dir/netblocks.txt" >> "$temp_dir/google-ipv4.txt"
 
-# save ipv4
-grep -v ':' /tmp/goog.txt > /tmp/google-ipv4.txt
-jq '.prefixes[] | [.ipv4Prefix][] | select(. != null)' -r /tmp/cloud.json >> /tmp/google-ipv4.txt
-jq '.prefixes[] | [.ipv4Prefix][] | select(. != null)' -r /tmp/googlebot.json >> /tmp/google-ipv4.txt
-grep -v ':' /tmp/netblocks.txt >> /tmp/google-ipv4.txt
+grep ':' "$temp_dir/goog.txt" > "$temp_dir/google-ipv6.txt"
+jq -r '.prefixes[] | select(.ipv6Prefix != null) | .ipv6Prefix' "$temp_dir/cloud.json" >> "$temp_dir/google-ipv6.txt"
+jq -r '.prefixes[] | select(.ipv6Prefix != null) | .ipv6Prefix' "$temp_dir/googlebot.json" >> "$temp_dir/google-ipv6.txt"
+grep ':' "$temp_dir/netblocks.txt" >> "$temp_dir/google-ipv6.txt"
 
-# save ipv6
-grep ':' /tmp/goog.txt > /tmp/google-ipv6.txt
-jq '.prefixes[] | [.ipv6Prefix][] | select(. != null)' -r /tmp/cloud.json >> /tmp/google-ipv6.txt
-jq '.prefixes[] | [.ipv6Prefix][] | select(. != null)' -r /tmp/googlebot.json >> /tmp/google-ipv6.txt
-grep ':' /tmp/netblocks.txt >> /tmp/google-ipv6.txt
+# Sort and deduplicate results, and ensure target directory exists
+output_dir="google"
+mkdir -p "$output_dir"
+sort -u "$temp_dir/google-ipv4.txt" > "$output_dir/ipv4.txt"
```

Comment:
`sort -V`:
>        -V, --version-sort
>              natural sort of (version) numbers within text


Without this sorting, it's gonna look bad, see https://github.com/lord-alfred/ipranges/issues/5 - Q3